spark/src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -562,6 +562,203 @@ class DeltaTable private[tables](
           TableFeatureProtocolUtils.FEATURE_PROP_SUPPORTED))
     toDataset(sparkSession, alterTableCmd)
   }
+
+  /**
+   * Clone a DeltaTable to a given destination to mirror the existing table's data and metadata.
+   *
+   * Specifying properties here means that the target will override any properties with the same key
+   * in the source table with the user-defined properties.
+   *
+   * An example would be
+   * {{{
+   *  io.delta.tables.DeltaTable.clone(
+   *   "/some/path",
+   *   true,
+   *   Map("foo" -> "bar"))
+   * }}}
+   *
+   * @param target The path or table name to create the clone.
+   * @param replace Whether to replace the destination with the clone command.
+   * @param properties The table properties to override in the clone.
+   * @since 3.3.0
+   */
+  def clone(target: String, replace: Boolean, properties: Map[String, String]): DeltaTable = {
+    executeClone(table, target, replace, properties)
+  }
+
+  /**
+   * Clone a DeltaTable to a given destination to mirror the existing table's data and metadata.
+   * 
+   * An example would be
+   * {{{
+   *  io.delta.tables.DeltaTable.clone("/some/path", true)
+   * }}}
+   *
+   * @param target The path or table name to create the clone.
+   * @param replace Whether to replace the destination with the clone command.
+   * @since 3.3.0
+   */
+  def clone(target: String, replace: Boolean): DeltaTable = {
+    clone(target, replace, properties = Map.empty)
+  }
+
+  /**
+   * Clone a DeltaTable to a given destination to mirror the existing table's data and metadata.
+   * 
+   * An example would be
+   * {{{
+   *  io.delta.tables.DeltaTable.clone("/some/path")
+   * }}}
+   *
+   * @param target The path or table name to create the clone.
+   *
+   * @since 3.3.0
+   */
+  def clone(target: String): DeltaTable = {
+    clone(target, replace = false)
+  }
+
+  /**
+   * Clone a DeltaTable at a specific version to a given destination to mirror the existing
+   * table's data and metadata.
+   *
+   * Specifying properties here means that the target will override any properties with the same key
+   * in the source table with the user-defined properties.
+   *
+   * An example would be
+   * {{{
+   *  io.delta.tables.DeltaTable.cloneAtVersion(
+   *   5,
+   *   "/some/path",
+   *   true,
+   *   Map("foo" -> "bar"))
+   * }}}
+   *
+   * @param version The version of this table to clone from.
+   * @param target The path or table name to create the clone.
+   * @param replace Whether to replace the destination with the clone command.
+   * @param properties The table properties to override in the clone.
+   *
+   * @since 3.3.0
+   */
+  def cloneAtVersion(
+      version: Long,
```

Comment:
Hmm, I was checking the SQL Clone code to make sure our Scala APIs codes are consistent with it.

Could we make `version` `Int`, looks like that's what the SQL code is doing as well.

https://github.com/delta-io/delta/blob/da5a5d29b6f00265439bc84020580080a2d4568f/spark/src/main/antlr4/io/delta/sql/parser/DeltaSqlBase.g4#L104

https://github.com/delta-io/delta/blob/da5a5d29b6f00265439bc84020580080a2d4568f/spark/src/main/antlr4/io/delta/sql/parser/DeltaSqlBase.g4#L134

https://github.com/delta-io/delta/blob/da5a5d29b6f00265439bc84020580080a2d4568f/spark/src/main/scala/io/delta/sql/parser/DeltaSqlParser.scala#L484

Reply:
`version` is tracked as a Long internally (https://github.com/delta-io/delta/blob/master/spark/src/main/scala/org/apache/spark/sql/delta/Snapshot.scala#L80), so if SQL is parsing it as an int that sounds like a SQL parser bug

Reply:
@Kimahriman Hmm, let me think more about that.

My additional thinking is if or if it is not a bug, then it has already been there and it is difficult to change to the behavior, and we need the Scala API to be consistent with SQL. I'm not sure if we will ever have a very large version also.

If it is a bug, do you think we can change the SQL Clone to accept version from Int to Long?

Reply:
Oh you already linked the line that shows it is treated as a long already

```
 Option(ctx.version).map(_.getText.toLong), 
```

Reply:
Yeah, but the input for SQL should be `IntegerType`

Reply:
Since Scala is quite type sensitive I don't think changing an existing type is easy

Reply:
The only difference is whether it looks for an `L` at the end of the SQL clause, it's not limiting it to just the size of a 32-bit integer

```g4
BIGINT_LITERAL
    : DIGIT+ 'L'
    ;

INTEGER_VALUE
    : DIGIT+
    ;
```

Reply:
Hmm I see, okay, `DIGIT+` seems like it means it accepts an arbitrary amount of digits.